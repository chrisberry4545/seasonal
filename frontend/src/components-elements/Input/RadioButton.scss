@import '../../styles/settings';

.c-radio-button {
  display: block;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  min-height: 30px;
  transition: background-color $default-animation-time-exit;

  &:hover {
    transition-duration: $default-animation-time-enter;
    background-color: transparentize($grey-light, 0.5);
  }
}

  .c-radio-button--selected {
    background-color: $grey-light;
  }

  .c-radio-button__input {
    display: none;
  }
