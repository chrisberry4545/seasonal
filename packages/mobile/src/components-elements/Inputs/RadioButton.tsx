import React, { FC } from 'react';
import { BareButton } from '../Buttons';
import { TextMedium } from '../Text';
import { TextStyle } from 'react-native';
import { styles } from '../../styles';

const styleRadioButton: TextStyle = {
  alignItems: 'center',
  display: 'flex',
  flex: 1,
  justifyContent: 'center',
  minHeight: 30
};
const styleRadioButtonSelected: TextStyle = {
  backgroundColor: styles.colors.selectionColor
};

export const RadioButton: FC<{
  isSelected?: boolean,
  label: string,
  onClick: () => void
}> = ({
  isSelected,
  label,
  onClick
}) => (
  <BareButton
    style={{
      ...styleRadioButton,
      ...(isSelected ? styleRadioButtonSelected : {})
    }}
    onClick={onClick}
    >
    <TextMedium>{label}</TextMedium>
  </BareButton>
);
