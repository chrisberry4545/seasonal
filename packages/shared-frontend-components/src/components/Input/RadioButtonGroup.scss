.c-radio-button-group {
  width: 100%;
  display: flex;
  justify-content: space-between;
}
