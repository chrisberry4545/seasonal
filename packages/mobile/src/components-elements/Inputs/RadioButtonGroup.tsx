import React, { FC } from 'react';

import { RadioButton } from './RadioButton';
import { View, ViewStyle } from 'react-native';

const styleRadioButtonGroup: ViewStyle = {
  display: 'flex',
  flexDirection: 'row'
};

export const RadioButtonGroup: FC<{
  radioButtons: Array<{
    value: string,
    label: string
  }>,
  selectedValue?: string,
  onChange: (newValue: string) => void,
  style?: ViewStyle;
}> = ({
  selectedValue,
  radioButtons,
  onChange,
  style
}) => (
  <View style={[styleRadioButtonGroup, style]}>
  {
    radioButtons.map(({ value, label }) => (
      <RadioButton
        key={value}
        isSelected={selectedValue === value}
        label={label}
        onClick={() => onChange(value)}
      />
    ))
  }
  </View>
);
