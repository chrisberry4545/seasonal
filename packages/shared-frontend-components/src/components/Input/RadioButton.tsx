import React, { FC } from 'react';
import './RadioButton.scss';

export const RadioButton: FC<{
  className?: string,
  defaultToChecked?: boolean,
  groupName: string,
  value: string,
  label: string
}> = ({
  className,
  defaultToChecked,
  groupName,
  value,
  label
}) => (
  <label
    data-e2e={`radio-btn-${value}`}
    htmlFor={value}
    className={`${(className || '')} c-radio-button${
      defaultToChecked ? ' c-radio-button--selected' : ''
    }`}>
    <input
      className='c-radio-button__input'
      defaultChecked={defaultToChecked}
      id={value}
      type='radio'
      name={groupName}
      value={value} />
    {label}
  </label>
);
