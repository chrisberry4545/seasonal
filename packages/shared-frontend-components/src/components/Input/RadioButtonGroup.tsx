import React, { FC } from 'react';

import { RadioButton } from './RadioButton';

import './RadioButtonGroup.scss';

export const RadioButtonGroup: FC<{
  className?: string,
  radioButtonClassName?: string,
  groupName: string,
  radioButtons: Array<{
    value: string,
    label: string
  }>,
  selectedValue?: string,
  onChange: (newValue: string) => void
}> = ({
  className,
  radioButtonClassName,
  selectedValue,
  groupName,
  radioButtons,
  onChange
}) => (
  <div
    className={`${(className || '')} c-radio-button-group`}
    onChange={
      (event) => onChange((event.target as HTMLInputElement).value)
    }>
    {
      radioButtons.map(({ value, label }) => (
        <RadioButton
          key={value}
          className={radioButtonClassName}
          defaultToChecked={selectedValue === value}
          groupName={groupName}
          value={value}
          label={label}
        />
      ))
    }
  </div>
);
