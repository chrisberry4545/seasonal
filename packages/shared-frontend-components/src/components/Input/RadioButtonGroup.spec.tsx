import React from 'react';
import { RadioButtonGroup } from './RadioButtonGroup';
import { shallow, ShallowWrapper } from 'enzyme';
import { RadioButton } from './RadioButton';

jest.mock('./RadioButton', () => ({
  RadioButton: () => 'RadioButton'
}));

describe('<RadioButtonGroup />', () => {
  let wrapper: ShallowWrapper;
  let mockOnChange: jest.Mock;
  const radioButtons: Array<{
    label: string,
    value: string
  }> = [
    {
      label: 'l1',
      value: 'v1'
    },
    {
      label: 'l2',
      value: 'v2'
    }
  ];

  beforeEach(() => {
    mockOnChange = jest.fn();
  });

  describe('when no values are selected', () => {
    beforeEach(() => {
      wrapper = shallow(
        <RadioButtonGroup onChange={mockOnChange}
          className='class'
          radioButtonClassName='radio-button-class'
          groupName='test-group'
          radioButtons={radioButtons} />
      );
    });

    test('renders correctly', () => expect(wrapper).toMatchSnapshot());

    test('outputs the value when changed', () => {
      wrapper.find('div').simulate('change', { target: { value: 'new-val' } });
      expect(mockOnChange).toHaveBeenCalledWith('new-val');
    });
  });

  describe('when a value is selected', () => {
    beforeEach(() => {
      wrapper = shallow(
        <RadioButtonGroup onChange={mockOnChange}
          className='class'
          radioButtonClassName='radio-button-class'
          groupName='test-group'
          radioButtons={radioButtons}
          selectedValue={radioButtons[0].value} />
      );
    });

    test('defaults to checked if the selected value matches the radio button value', () =>
      expect(wrapper.find(RadioButton).first().prop('defaultToChecked'))
        .toBe(true));

    test('does not defaults to checked if the selected value does not matches the radio button value', () =>
      expect(wrapper.find(RadioButton).last().prop('defaultToChecked'))
        .toBe(false));
  });

});
