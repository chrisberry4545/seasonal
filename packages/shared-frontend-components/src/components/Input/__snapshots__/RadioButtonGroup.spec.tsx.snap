// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButtonGroup /> when no values are selected renders correctly 1`] = `
<div
  className="class c-radio-button-group"
  onChange={[Function]}
>
  <RadioButton
    className="radio-button-class"
    defaultToChecked={false}
    groupName="test-group"
    key="v1"
    label="l1"
    value="v1"
  />
  <RadioButton
    className="radio-button-class"
    defaultToChecked={false}
    groupName="test-group"
    key="v2"
    label="l2"
    value="v2"
  />
</div>
`;
